### ISSUE TITLE
Screensaver segfaulted after wakeup from notebook standby

### ISSUE THREAD

**Issue by qwertfisch** [linuxmint/cinnamon-screensaver#371]
### Hardware Information
 * Cinnamon-Screensaver version: 4.8.1
 * Linux Mint 20.1 (fresh installation, no upgrade)
 * Kernel version: 5.4.0-66-generic
 * VGA compatible controller: Intel Corporation 2nd Generation Core Processor Family Integrated Graphics Controller (rev 09)
 * Kernel driver in use: i915
 * 64 bit

### Issue
After opening the notebook lid and waking it up from standby, I was typing my user password blindly (because the display becomes active only ~1,5s after opening). Before I finished typing I saw there was no lockscreen active and I did not need to enter the password, even though the notebook locks whenever I close the notebook lid.

A quick bug search in the systemd journal showed me the output attached to this issue. The process cinnamon-screen segfaulted (line 27), then there is the core dump listed. See [journald-crash.log](https://github.com/linuxmint/cinnamon-screensaver/files/6108604/journald-crash.log)

**Steps to reproduce**
Unfortunately I cannot reproduce this behaviour. It occurred once in November and then again three days ago.

**Expected behaviour**
The screensaver / lockscreen should not crash, and if it crashes should be restarted or otherwise prevent using the computer without authentication.

**Other information**
If some information is missing or might be helpful to investigate the issue, I will be happy to provide it.

**Comment by clefebvre** [comment]
@mtwebster do you see anything relevant in the stack?

Aside from this particular crash, we have an item in the roadmap to 5.0. We want to make sure the screensaver crash doesn't result in access to the session. An issue between X11 and libcaribou earlier this year highlighted the need for this. We definitely need to have this in place in 5.0.

**Comment by mtwebster** [comment]
Hi, since you're in mint, can you open 'System Reports' and select the 'Crash reports' view. Hopefully there you'll find an entry for cinnamon-screensaver (or it may say cinnamon-screensaver-pam-helper) - if there is one, click the 'Local files' button and paste crash.tar.gz here - I should be able to reconstruct it and get something useful.

Thanks

**Comment by mtwebster** [comment]
Also, do you have any sort of additional methods of authentication setup? Such as a fingerprint reader or yubikey?

**Comment by qwertfisch** [comment]
> Hopefully there you'll find an entry for cinnamon-screensaver (or it may say cinnamon-screensaver-pam-helper) - if there is one, click the 'Local files' button and paste crash.tar.gz here

@mtwebster Yes, there was a crash report for the cinnamon-screensaver-pam-helper. Please have a look at the 
[crash.tar.gz](https://github.com/linuxmint/cinnamon-screensaver/files/6109775/crash.tar.gz) here. I hope 1.2MB is not too big.


> Also, do you have any sort of additional methods of authentication setup? Such as a fingerprint reader or yubikey?

No, there is only one user setup with a plain password. No other (hardware) authentication is used.

**Comment by mtwebster** [comment]
Hi, were there any more dumps in the journal (or in Mint Report)? I'm pretty sure the pam-helper crashing is just collateral damage from the real screensaver process crashing..I'm not sure why that wasn't captured as well in the journal other than a single line:
```
Mär 06 20:11:52 shuttle kernel: cinnamon-screen[9891]: segfault at 100000001 ip 00007f5ab55ea2ed sp 00007ffc82cc5a20 error 4 in libgdk-3.so.0.2404.16[7f5ab55e0000+81000]
```

Could I get you to install the debug symbols for a few libraries - this might enable us to get a bit more info if it happens again:
```
apt install libgtk-3-0-dbgsym libgdk-pixbuf2.0-0-dbgsym libglib2.0-0-dbgsym libcscreensaver-dbg
```
You'll need to open 'Software Sources' and turn on 'Debug symbols' first.

Thanks

**Comment by qwertfisch** [comment]
I installed the debug symbols as suggested, but as the error is quite rare, we might never get a debug stack trace.

There was only one other crash report in the list, from several hours before, but it’s related to guvcview (just testing my webcam capabilities), and I at least once did suspend and wakeup the notebook successfully before the issue.

In the attached journal log from the initial post I can see **two** core dumps, both from cinnamon-screen (one with PID 57447, one with PID 9891). The Mint system report app only lists the first one, but I found in `/var/lib/systemd/coredump` that both core dumps were written. I have attached it here as well, if it helps? → [core.cinnamon-screen.1000.9891.tar.gz](https://github.com/linuxmint/cinnamon-screensaver/files/6114963/core.cinnamon-screen.1000.9891.tar.gz)

**Comment by admksh** [comment]
I also noticed this issue, multiple times now. It does not happen often, but since I keep my laptop on standby for long periods of time, I probably see it more often than @qwertfisch does.

I've attached the relevant crash report and system log with this post. Unfortunately, I did not enable debug symbols beforehand.

[crash.tar.gz](https://github.com/linuxmint/cinnamon-screensaver/files/6447013/crash.tar.gz) - [journal.log](https://github.com/linuxmint/cinnamon-screensaver/files/6447014/journal.log)

---

## System Information

- Linux Mint 20.1 Ulyssa (up-to-date)
- Linux 5.4.0-72-generic
- Cinnamon 4.8.6
- cinnamon-screensaver 4.8.1

**Comment by qwertfisch** [comment]
@mtwebster After more than two months, the initially reported issue happened again. I still have the debug symbols installed, so there might be a better chance in finding anything?

Files attached:
- [crash.tar.gz](https://github.com/linuxmint/cinnamon-screensaver/files/6499751/crash.tar.gz)
- [journald crashlog](https://github.com/linuxmint/cinnamon-screensaver/files/6499772/journal_2021-05-18.log) from closing notebook lid (= suspend) to reopening. This lists two consecutive crashes (only the second one is listed in Mint crash reports). If you are also interested in the raw coredump (8MB compressed, 428MB uncompressed), let me know.

**Comment by kelebek333** [comment]
Cinnamon-Screensaver version: 5.0.5
Linux Mint 20.2
Kernel version: 5.12.10-xanmod1
VGA compatible controller: Nvidia 9500 GT
Kernel driver in use: nvidia-340

Same issue after suspend on Linux Mint 20.2 uma. Crash report => https://paste2.org/fbaYkwmt

**Comment by mtwebster** [comment]
@kelebek do you just have the laptop monitor? (no external). When the crash happened were you left with a fully unlocked desktop? Or did you get a black screen with instructions on how to unlock?

**Comment by kelebek333** [comment]
@mtwebster ,  I am using a desktop computer, not a laptop. 

I got a black screen with instructions on how to unlock.
